node/processingOnlyNode/cryptoComponents.go

```diff
@@ -1,3 +1,240 @@
+func (c *cryptoComponentsHolder) PublicKeyBytes() []byte {
+	return c.publicKeyBytes
+}
+
+// P2pPublicKey will return the p2p public key
+func (c *cryptoComponentsHolder) P2pPublicKey() crypto.PublicKey {
+	return c.p2pPublicKey
+}
+
+// P2pPrivateKey will return the p2p private key
+func (c *cryptoComponentsHolder) P2pPrivateKey() crypto.PrivateKey {
+	return c.p2pPrivateKey
+}
+
+// P2pSingleSigner will return the p2p single signer
+func (c *cryptoComponentsHolder) P2pSingleSigner() crypto.SingleSigner {
+	return c.p2pSingleSigner
+}
+
+// TxSingleSigner will return the transaction single signer
+func (c *cryptoComponentsHolder) TxSingleSigner() crypto.SingleSigner {
+	return c.txSingleSigner
+}
+
+// BlockSigner will return the block signer
+func (c *cryptoComponentsHolder) BlockSigner() crypto.SingleSigner {
+	return c.blockSigner
+}
+
+// SetMultiSignerContainer will set the multi signer container
+func (c *cryptoComponentsHolder) SetMultiSignerContainer(container cryptoCommon.MultiSignerContainer) error {
+	c.multiSignerContainer = container
+
+	return nil
+}
+
+// MultiSignerContainer will return the multi signer container
+func (c *cryptoComponentsHolder) MultiSignerContainer() cryptoCommon.MultiSignerContainer {
+	return c.multiSignerContainer
+}
+
+// GetMultiSigner will return the multi signer by epoch
+func (c *cryptoComponentsHolder) GetMultiSigner(epoch uint32) (crypto.MultiSigner, error) {
+	return c.MultiSignerContainer().GetMultiSigner(epoch)
+}
+
+// PeerSignatureHandler will return the peer signature handler
+func (c *cryptoComponentsHolder) PeerSignatureHandler() crypto.PeerSignatureHandler {
+	return c.peerSignatureHandler
+}
+
+// BlockSignKeyGen will return the block signer key generator
+func (c *cryptoComponentsHolder) BlockSignKeyGen() crypto.KeyGenerator {
+	return c.blockSignKeyGen
+}
+
+// TxSignKeyGen will return the transaction sign key generator
+func (c *cryptoComponentsHolder) TxSignKeyGen() crypto.KeyGenerator {
+	return c.txSignKeyGen
+}
+
+// P2pKeyGen will return the p2p key generator
+func (c *cryptoComponentsHolder) P2pKeyGen() crypto.KeyGenerator {
+	return c.p2pKeyGen
+}
+
+// MessageSignVerifier will return the message signature verifier
+func (c *cryptoComponentsHolder) MessageSignVerifier() vm.MessageSignVerifier {
+	return c.messageSignVerifier
+}
+
+// ConsensusSigningHandler will return the consensus signing handler
+func (c *cryptoComponentsHolder) ConsensusSigningHandler() consensus.SigningHandler {
+	return c.consensusSigningHandler
+}
+
+// ManagedPeersHolder will return the managed peer holder
+func (c *cryptoComponentsHolder) ManagedPeersHolder() common.ManagedPeersHolder {
+	return c.managedPeersHolder
+}
+
+// KeysHandler will return the keys handler
+func (c *cryptoComponentsHolder) KeysHandler() consensus.KeysHandler {
+	return c.keysHandler
+}
+
+// Clone will clone the cryptoComponentsHolder
+func (c *cryptoComponentsHolder) Clone() interface{} {
```

Comment:
why not return a `*cryptoComponentsHolder` ?

Reply:
I had to implement the interface